### PR TITLE
Remove default admin credentials

### PR DESCRIPTION
**Pull request by jackiehanyang**
### Description
Remove default admin credentials `admin:admin`

### Issues Resolved
https://github.com/opensearch-project/anomaly-detection/issues/1120

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: .github/workflows/test_security.yml [hunk 1]
```diff
@@ -72,11 +72,11 @@ jobs:
       - name: Run AD Test
         if: env.imagePresent == 'true'
         run: |
-          security=`curl -XGET https://localhost:9200/_cat/plugins?v -u admin:admin --insecure |grep opensearch-security|wc -l`
+          security=`curl -XGET https://localhost:9200/_cat/plugins?v -u admin:myStrongPassword123! --insecure |grep opensearch-security|wc -l`
```
Comment (DarshitChanpura) [inline .github/workflows/test_security.yml:75]: this password needs to be passed to the containers on line 70 with `-e OPENSEARCH_INITIAL_ADMIN_PASSWORD=myStrongPassword123!`

There is no default password for admin user starting 2.12. This password needs to be set, and you can set it as above.

Also, this change should only be present for branches pointing to 2.12 image.

File: DEVELOPER_GUIDE.md [hunk 2]
```diff
@@ -42,7 +42,7 @@ Currently we just put RCF jar in lib as dependency. Plan to publish to Maven and
 2. `./gradlew :run` launches a single node cluster with anomaly-detection (and job-scheduler) plugin installed
 3. `./gradlew :integTest` launches a single node cluster with anomaly-detection (and job-scheduler) plugin installed and runs all integration tests except security
 4. ` ./gradlew :integTest --tests="**.test execute foo"` runs a single integration test class or method
-5. `./gradlew integTestRemote -Dtests.rest.cluster=localhost:9200 -Dtests.cluster=localhost:9200 -Dtests.clustername="docker-cluster" -Dhttps=true -Duser=admin -Dpassword=admin` launches integration tests against a local cluster and run tests with security
+5. `./gradlew integTestRemote -Dtests.rest.cluster=localhost:9200 -Dtests.cluster=localhost:9200 -Dtests.clustername="docker-cluster" -Dhttps=true -Duser=admin -Dpassword=myStrongPassword123!` launches integration tests against a local cluster and run tests with security
```
Comment (DarshitChanpura) [inline DEVELOPER_GUIDE.md:45]: `myStrongPassword123!` should not be considered as default password.

`admin` should be replaced with a placeholder `<admin-password>`

File: dataGeneration/README.md [hunk 3]
```diff
@@ -51,7 +51,7 @@ The dataset created will have two categorical fields to test a multi-entity AD (
 | --number-of-process | number of 'process' entities (process is one of the categorical field that an entity is defined by)| 1000 | No
 | --number-of-historical-days | number of day of historical data to ingest | 2 | No
 | --username | username for authentication if security is true | admin | No
-| --password | password for authentication if security is true | admin | No
+| --password | password for authentication if security is true | myStrongPassword123! | No
```
Comment (DarshitChanpura) [inline dataGeneration/README.md:54]: same as above

File: dataGeneration/generate-cosine-data-multi-entity.py [hunk 4]
```diff
@@ -42,7 +42,7 @@
 parser.add_argument("-np", "--number-of-process", type=int, default=1000, help="number of 'process' entities, deafult is set to 1000, there will be two keyword categories in this index (must be at least 1)" )
 parser.add_argument("-hd", "--number-of-historical-days", type=int, default=2, help="number of day of historical data to ingest, defaults to 2")
 parser.add_argument("-u", "--username", type=str, default="admin", help="username for authentication if security is true")
-parser.add_argument("-pass", "--password", type=str, default="admin", help="password for authentication if security is true")
+parser.add_argument("-pass", "--password", type=str, default="myStrongPassword123!", help="password for authentication if security is true")
```
Comment (DarshitChanpura) [inline dataGeneration/generate-cosine-data-multi-entity.py:45]: same as above

**Review by DarshitChanpura — CHANGES_REQUESTED** [review]
Ty for making this change. There are couple more changes required before this is ready.

**Comment by joshpalis** [comment]
Seems the security enabled integration tests are being skipped all together due to the cluster not being available, not due to the security plugin being unavailable ([reference run](https://github.com/opensearch-project/anomaly-detection/actions/runs/7619224062/job/20751910190?pr=1134)) : 
```
Run security=`curl -XGET https://localhost:9[20](https://github.com/opensearch-project/anomaly-detection/actions/runs/7619224062/job/20751910190?pr=1134#step:7:21)0/_cat/plugins?v -u admin:myStrongPassword123! --insecure |grep opensearch-security|wc -l`
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed

  0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0
  0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0
curl: (7) Failed to connect to localhost port 9200 after 0 ms: Connection refused
Security plugin is NOT available, skipping integration tests
```

@jackiehanyang This is due to issues with the provided docker image, but can you confirm if these changes pass locally?

File: .github/workflows/test_security.yml (hunk 1 above)
Comment (jackiehanyang) [reply .github/workflows/test_security.yml:75]: > Also, this change should only be present for branches pointing to 2.12 image.

Should this change be present for branches pointing to 2.12 image or 2.12 and above image?

File: DEVELOPER_GUIDE.md (hunk 2 above)
Comment (jackiehanyang) [reply DEVELOPER_GUIDE.md]: Updated with placeholder `<admin-password>` in the latest revision

File: dataGeneration/README.md (hunk 3 above)
Comment (jackiehanyang) [reply dataGeneration/README.md]: Updated with placeholder `<admin-password>` in the latest revision

File: dataGeneration/generate-cosine-data-multi-entity.py (hunk 4 above)
Comment (jackiehanyang) [reply dataGeneration/generate-cosine-data-multi-entity.py:45]: Updated with placeholder `<admin-password>` in the latest revision in all README files. Leaving it as `myStrongPassword123!` in this file as it's not a README file

**Comment by jackiehanyang** [comment]
> Seems the security enabled integration tests are being skipped all together due to the cluster not being available, not due to the security plugin being unavailable ([reference run](https://github.com/opensearch-project/anomaly-detection/actions/runs/7619224062/job/20751910190?pr=1134)) :
> 
> ```
> Run security=`curl -XGET https://localhost:9[20](https://github.com/opensearch-project/anomaly-detection/actions/runs/7619224062/job/20751910190?pr=1134#step:7:21)0/_cat/plugins?v -u admin:myStrongPassword123! --insecure |grep opensearch-security|wc -l`
>   % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
>                                  Dload  Upload   Total   Spent    Left  Speed
> 
>   0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0
>   0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0
> curl: (7) Failed to connect to localhost port 9200 after 0 ms: Connection refused
> Security plugin is NOT available, skipping integration tests
> ```
> 
> @jackiehanyang This is due to issues with the provided docker image, but can you confirm if these changes pass locally?

This is a known issue for AD. The tests were passing locally and there aren't much new changes for AD lately. We should be fine here. Cc @kaituo

File: .github/workflows/test_security.yml (hunk 1 above)
Comment (DarshitChanpura) [reply .github/workflows/test_security.yml:75]: The change should be present starting from 2.12. (not in 2.11.1 and below).

> This change is towards main branch and will be backported to 2.x branch

yes.

**Review by DarshitChanpura — APPROVED** [review]
LGTM. Thanks @jackiehanyang !

**Review by ohltyler — APPROVED** [review]

**Pull request merged**